torch/_inductor/codegen/triton.py

```diff
@@ -2571,3 +2573,132 @@ def store_cache():
         )
         store_cache()
         return ms, mod.__file__
+
+    def benchmark_combo_kernel(self, node_list):
+        from .triton_combo_kernel import ComboKernel
+
+        def cache_file_path():
+            assert mod.__file__ is not None
+            return os.path.splitext(mod.__file__)[0] + ".kernel_perf"
+
+        def load_cache():
+            path = cache_file_path()
+            if os.path.exists(path):
+                with open(path) as fd:
+                    return tuple(float(e) for e in fd.read().split())
+            return (None, None)
+
+        def store_cache():
+            path = cache_file_path()
+            with open(path, "w") as fd:
+                fd.write(str(ms) + " " + str(ms_clone))
+
+        subkernel_nodes = node_list
+        fused_node_lists = [node.get_nodes() for node in subkernel_nodes]
+        subkernel_map, node_schedule_map = {}, {}
+        for pn, nodes in zip(subkernel_nodes, fused_node_lists):
+            _, (numel, rnumel) = max(nodes, key=lambda x: int(x.is_reduction())).group
```

Comment:
Can we refactor out a lot of this duplicated logic ? E.g. for `codegen_template` i added `only_gen_src_code` arg

Reply:
I see. will work on it.